aut-shuttle/frontend
# Qr reader implementation

Still a work in progress, PR'd due to people needing this element for the driver dashboard etc. It will allow them to integrate my code into their code, rather than trying to implement it after its completed.

QRReader uses computer webcam

**How to Test**
Log in as a user
Generate a QR code
Photograph or print QR code
Open QR reader
Scan QR code
Add passenger to bus
(Repeat for extra passengers - I have a photo of 5 different QR codes)
Click "Click here when ready to leave"
Check console - "Bus" array is there

**Completed**
Core scanning functionality
Asynchronous interaction with state
UI and information provided to Driver up scanning
Balance checking

**Remaining Work**
Requires error checking
Addition of API route for trip and boarding
